entry_types/scrolled/package/src/frontend/inlineEditing/SharingTooltip.js

```diff
@@ -0,0 +1,64 @@
+import React, {useEffect, useState} from 'react';
+import classNames from 'classnames';
+import headerStyles from "../navigation/AppHeader.module.css";
+import styles from "../navigation/SharingTooltip.module.css";
+import ReactTooltip from "react-tooltip";
+import {useShareProviders, useShareUrl} from '../../entryState';
+
+export function SharingTooltip() {
+
+  const [phoneEmulationMode, setPhoneEmulationMode] = useState(false);
+  const shareUrl = useShareUrl();
+
+  useEffect(() => {
+    window.addEventListener('message', receive);
+
+    function receive(event) {
+      if (event.data.type === 'CHANGE') {
+        if(event.data['payload'] === 'phone') {
+          setPhoneEmulationMode(true);
+        }
+        else {
+          setPhoneEmulationMode(false);
+        }
+      }
+    }
+    return () => document.removeEventListener('message', receive);
+  });
+
+  const shareProviders = useShareProviders(phoneEmulationMode);
+
+  function renderShareLinks(shareProviders) {
+    return shareProviders.map((shareProvider) => {
+      const Icon = shareProvider.icon;
+      return (
+        <div key={shareProvider.name}
+             className={styles.shareLinkContainer}>
+          <a className={classNames('share', styles.shareLink)}
+             href={shareProvider.url.replace('%{url}', shareUrl)}
+             target={'_blank'}>
+            <Icon className={styles.shareIcon}/>
+            {shareProvider.name}
+          </a>
+        </div>
+      )
+    })
+  };
+
+  return (
+    <ReactTooltip id={'sharingTooltip'}
+                  type={'light'}
+                  place={'bottom'}
+                  effect={'solid'}
+                  event={'click'}
+                  globalEventOff={'click'}
+                  clickable={true}
+                  offset={{right: -64}}
+                  className={classNames(headerStyles.navigationTooltip,
+                                        styles.sharingTooltip)}>
+      <div onMouseLeave={() => { ReactTooltip.hide() }}>
+        {renderShareLinks(shareProviders)}
+      </div>
+    </ReactTooltip>
+  )
+}
```

Comment:
Instead of replacing the `SharingTooltip` with an inline editing specific component, I'd suggest

* introducing a `PhonePlatformProvider` component in `src/frontend` which calls `browser.has` and renders the `PhonePlatformContext.Provider` 
* adding an inline editing replacement for `PhonePlatformProvider` which contains the above post message listening code and state and renders a `PhonePlatformContext.Provider` based on that state.
* defining a `usePhonePlatform` hook that is a wrapper for `useContext(PhonePlatformContext)`

This has a couple of advantages:

* No need to duplicate the code of `SharingTooltip`
* `usePhonePlatform` can then be used freely in any component and will work correctly in inline editing mode. With the current approach, we would have to create an inline editing alternative of every component that is supposed to depend on phone platform.

Reply:
Changed the logic according to the suggestion. Please, see the commit.